hal/src/main/native/sim/MockData/DriverStationDataInternal.h

```diff
@@ -96,6 +106,10 @@ class DriverStationData {
   std::shared_ptr<NotifyListenerVector> m_allianceStationIdCallbacks = nullptr;
   std::atomic<double> m_matchTime{0.0};
   std::shared_ptr<NotifyListenerVector> m_matchTimeCallbacks = nullptr;
+
+  std::map<int, HAL_JoystickAxes> m_joystickAxis;
+  std::map<int, HAL_JoystickPOVs> m_joystickPov;
+  std::map<int, HAL_JoystickButtons> m_joystickButtons;
```

Comment:
std::map doesn't feel like the right thing to use here.  Why not std::vector?

Reply:
I could do that.  Using the map and "initializing" the structs on the Set* function saves me the having to preallocate the vector and zero out all the fields.  But I guess I didn't go that way out of laziness.